src/source/oscap_source.c

```diff
@@ -363,7 +363,7 @@ const char *oscap_source_get_schema_version(struct oscap_source *source)
 		}
 		switch (oscap_source_get_scap_type(source)) {
 			case OSCAP_DOCUMENT_SDS:
-				source->origin.version = oscap_strdup("1.2");
+				source->origin.version = ds_sds_detect_version(reader);
```

Comment:
Can we add a simple test for this? It should be sufficient to use `oscap info` which uses the `oscap_source_get_schema_version` function and call it against prepared content, one with `schematron-version` `1.2 and the other with `1.3` and check if the versions are reported correctly.